docs/components/zeebe/technical-concepts/batch-operations.md

```diff
@@ -6,117 +6,424 @@ description: How distributed batch operations work in Zeebe engine.
 
 Zeebe supports executing certain operations as batch operations:
 
-- Resolve incidents
-- Migrate process instances
-- Modify process instances
-- Cancel process instances
+- **Cancel process instances** (`CANCEL_PROCESS_INSTANCE`)
+- **Migrate process instances** (`MIGRATE_PROCESS_INSTANCE`)
+- **Modify process instances** (`MODIFY_PROCESS_INSTANCE`)
+- **Resolve incidents** (`RESOLVE_INCIDENT`)
 
-## How do they work?
+## Overview
+
+Batch operations allow you to perform actions on multiple process instances efficiently. This is particularly useful when you need to:
+
+- Cancel hundreds of process instances after discovering a critical bug
+- Migrate thousands of running instances to a new process version
+- Modify process instances to skip or add activities across multiple instances
+- Resolve incidents across multiple process instances simultaneously
+
+Instead of manually operating on each instance, batch operations distribute the work across your cluster, processing instances in parallel while maintaining system stability.
+
+### How it works at a high level
+
+When you create a batch operation, you provide filter criteria (such as process definition, version, or state) rather than a specific list of instances. The system then:
+
+1. Distributes the operation across all partitions in your cluster
+2. Each partition queries the secondary database to find matching instances
+3. Processes instances in parallel across partitions
+4. Tracks progress and aggregates results
+
+This distributed approach enables efficient processing of large numbers of instances while maintaining cluster performance.
+
+## Prerequisites
+
+To use batch operations, you need:
+
+- A Zeebe cluster with secondary storage configured (Elasticsearch or OpenSearch)
+- Appropriate [authorization permissions](#authorization) for the operations you want to perform
+- Access to one of the following:
+  - [Operate UI](/components/operate/operate-introduction.md) (for basic operations)
+  - [Orchestration Cluster REST API](/apis-tools/orchestration-cluster-api-rest/orchestration-cluster-api-rest-overview.md) (for full lifecycle management)
+  - [Camunda Java client](/apis-tools/java-client/getting-started.md) (for programmatic access)
```

Comment:
🔧 This appears like a comprehensive list, but it isn't and that would be hard to maintain in this page. I suggest replacing `Camunda Java client` with an API Client (simply linking to [APIs & tools](https://preview.docs.camunda.cloud/pr-7370/docs/apis-tools/working-with-apis-tools/)